classes/services/PKPSubmissionService.inc.php

```diff
@@ -941,4 +941,66 @@ public function updateStatus($submission)
 
         return $submission;
     }
+
+    /**
+     * Checks if this user is granted reader access to pre-publication submissions
+     * based on their roles in the context (i.e. Manager, Editor, etc).
+     *
+     * @param $user User
+     * @param $submission Submission
+     *
+     * @return bool
+     */
+    public function canPreview($user, $submission)
+    {
+        // Don't grant access until submission reaches Copyediting stage
+        if ($submission->getData('stageId') < WORKFLOW_STAGE_ID_EDITING) {
```

Comment:
I don't recommend using `<>` comparisons on defined constants -- the numerical values of these aren't intended to have significance beyond being unique. Better to use `in_array` and list the options, I think.